fe/src/main/java/org/apache/doris/load/loadv2/BitmapValue.java

```diff
@@ -0,0 +1,372 @@
+                        this.bitmapType = BITMAP_VALUE;
+                        break;
+                }
+                break;
+        }
+    }
+
+    public boolean equals(BitmapValue other) {
+        boolean ret = false;
+        switch (bitmapType) {
+            case EMPTY:
+                switch (other.bitmapType) {
+                    case EMPTY:
+                        ret = true;
+                        break;
+                    case SINGLE_VALUE:
+                    case BITMAP_VALUE:
+                        ret = false;
+                        break;
+                }
+                break;
+            case SINGLE_VALUE:
+                switch (other.bitmapType) {
+                    case EMPTY:
+                    case BITMAP_VALUE:
+                        break;
+                    case SINGLE_VALUE:
+                        ret = this.singleValue == other.singleValue;
+                        break;
+                }
+                break;
+            case BITMAP_VALUE:
+                switch (other.bitmapType) {
+                    case EMPTY:
+                    case SINGLE_VALUE:
+                        ret = false;
+                        break;
+                    case BITMAP_VALUE:
+                        ret = bitmap.equals(other.bitmap);
+                        break;
+                }
+                break;
+        }
+        return ret;
+    }
+
+    public long getSizeInBytes() {
+        long size = 0;
+        switch (bitmapType) {
+            case EMPTY:
+                size = 1;
+                break;
+            case SINGLE_VALUE:
+                if (singleValue <= Integer.MAX_VALUE) {
+                    size = 4;
+                } else {
+                    size = 8;
+                }
+                break;
+            case BITMAP_VALUE:
+                size = bitmap.getSizeInBytes();
+        }
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        String toStringStr = "{}";
+        switch (bitmapType) {
+            case EMPTY:
+                break;
+            case SINGLE_VALUE:
+                toStringStr = String.format("{%s}", singleValue);
+                break;
+            case BITMAP_VALUE:
+                toStringStr = this.bitmap.toString();
+                break;
+        }
+        return toStringStr;
+    }
+
+    public void clear() {
+        this.bitmapType = EMPTY;
+        this.bitmap = null;
+    }
+
+    private void convertToSmallerType() {
+        if (bitmapType == BITMAP_VALUE) {
+            if (bitmap.getLongCardinality() == 0) {
+                this.bitmap = null;
+                this.bitmapType = EMPTY;
+            } else if (bitmap.getLongCardinality() == 1) {
+                this.singleValue = bitmap.select(0);
+                this.bitmapType = SINGLE_VALUE;
+                this.bitmap = null;
+            }
+        }
+    }
+
+    private boolean isLongValue32bitEnough(long value) {
```

Comment:
4294967293 will return false.

which is not consistent with is32BitsEnough in BE
```
bool is32BitsEnough() const { return maximum() <= std::numeric_limits<uint32_t>::max(); }
```

Reply:
In java,32-bit integers can be represented and used with arithmetic operation using long.
But support 64-bit unsigned integer is really a complex thing in java system.
So here I think it's ok to support unsigned 32-bit integers in FE's bitmap.
But I prefer 64-bit unsigned integer not being considered for the time being.

Reply:
OK